libs/grok/src/main/java/org/elasticsearch/grok/Grok.java

```diff
@@ -252,7 +255,7 @@ public boolean match(String text) {
      * @param length length of the text to match
      * @return a map containing field names and their respective coerced values that matched.
      */
-    public Map<String, Object> captures(byte[] utf8Bytes, int offset, int length) {
+    public boolean match(byte[] utf8Bytes, int offset, int length, GrokCaptureExtracter extracter) {
```

Comment:
Comment should be updated, this is return boolean on success now and mutating the `extracter`